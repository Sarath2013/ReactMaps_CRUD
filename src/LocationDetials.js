import React from 'react';
import PropTypes from 'prop-types'

const LocationDetials = (props) => {
  return (
    <div>
      <div className='loc-title'>{props.location.name}</div>
      <div className='loc-name'>{props.location.name}</div>
      <div className='loc-lat'>latitude: {props.location.lat}</div>
      <div className='loc-lng'>longitude: {props.location.lng}</div>
    </div>
  );
}

LocationDetials.propTypes = {
  location: PropTypes.shape({
      name: PropTypes.string.isRequired,
      lat: PropTypes.number.isRequired,
      lng: PropTypes.number.isRequired
    }).isRequired,
}

export default LocationDetials;