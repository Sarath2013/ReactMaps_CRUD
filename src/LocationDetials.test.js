import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LocationDetials from './LocationDetials';

const props = {
  location: {
    name: 'Berlin',
    lat: 52.52,
    lng: 13.405
  }
};

let wrapper;

describe("Rendering LocaitonDetials", () => {
  beforeAll(() => {
    wrapper = shallow(<LocationDetials {...props}/>);
  });
  
  afterAll(() => {
    wrapper = null
  });
  
  test('renders snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('ensure location detials', () => {
    expect(wrapper.find('.loc-title').text()).toEqual('Berlin');
    expect(wrapper.find('.loc-name').text()).toEqual('Berlin');
    expect(wrapper.find('.loc-lat').text()).toEqual('latitude: 52.52');
    expect(wrapper.find('.loc-lng').text()).toEqual('longitude: 13.405');
  });
});


