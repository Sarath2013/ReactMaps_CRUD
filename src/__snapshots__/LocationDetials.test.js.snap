// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering LocaitonDetials renders snapshot 1`] = `
<div>
  <div
    className="loc-title"
  >
    Berlin
  </div>
  <div
    className="loc-name"
  >
    Berlin
  </div>
  <div
    className="loc-lat"
  >
    latitude: 
    52.52
  </div>
  <div
    className="loc-lng"
  >
    longitude: 
    13.405
  </div>
</div>
`;
